mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -592,6 +629,59 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
   }
 
   /**
+    * Tweedie exponential family distribution.
+    * The default link for the Tweedie family is the log link.
+    */
+  private[regression] object Tweedie extends Family("tweedie") {
+
+    val defaultLink: Link = Log
+
+    var variancePower: Double = 1.5
+
+    override def initialize(y: Double, weight: Double): Double = {
+      if (variancePower > 1.0 && variancePower < 2.0) {
+        require(y >= 0.0, "The response variable of the specified Tweedie distribution " +
+          s"should be non-negative, but got $y")
+        math.max(y, 0.1)
```

Comment:
If we're going to use this magic 0.1 constant in many places, factor out a constant? 0.1 seems quite large as an 'epsilon' but I guess that's what R's implementation uses for whatever reason?